core/src/main/java/org/apache/carbondata/core/datastore/page/encoding/RLECodec.java

```diff
@@ -0,0 +1,417 @@
+    // NONREPEATED_RUN state means it is collecting non-repeated values (`nonRepeatValues`)
+    private RUN_STATE runState;
+
+    // count for each run, either REPEATED_RUN or NONREPEATED_RUN
+    private int valueCount;
+
+    // collected value for REPEATED_RUN
+    private Object lastValue;
+
+    // collected value for NONREPEATED_RUN
+    private List<Object> nonRepeatValues;
+
+    // data type of input page
+    private DataType dataType;
+
+    // output stream for encoded data
+    private ByteArrayOutputStream bao;
+    private DataOutputStream stream;
+
+    private Encoder() {
+      this.runState = RUN_STATE.INIT;
+      this.valueCount = 0;
+      this.nonRepeatValues = new ArrayList<>();
+      this.bao = new ByteArrayOutputStream();
+      this.stream = new DataOutputStream(bao);
+    }
+
+    private EncodedColumnPage encode(ColumnPage input) throws MemoryException, IOException {
+      validateDataType(input.getDataType());
+      this.dataType = input.getDataType();
+      switch (dataType) {
+        case BYTE:
+          byte[] bytePage = input.getBytePage();
+          for (int i = 0; i < bytePage.length; i++) {
+            putValue(bytePage[i]);
+          }
+          break;
+        case SHORT:
+          short[] shortPage = input.getShortPage();
+          for (int i = 0; i < shortPage.length; i++) {
+            putValue(shortPage[i]);
+          }
+          break;
+        case INT:
+          int[] intPage = input.getIntPage();
+          for (int i = 0; i < intPage.length; i++) {
+            putValue(intPage[i]);
+          }
+          break;
+        case LONG:
+          long[] longPage = input.getLongPage();
+          for (int i = 0; i < longPage.length; i++) {
+            putValue(longPage[i]);
+          }
+          break;
+        default:
+          throw new UnsupportedOperationException(input.getDataType() +
+              " does not support RLE encoding");
+      }
+      byte[] encoded = collectResult();
+      SimpleStatsResult stats = (SimpleStatsResult) input.getStatistics();
+      return new EncodedMeasurePage(
+          input.getPageSize(),
+          encoded,
+          CodecMetaFactory.createMeta(stats, input.getDataType()),
+          stats.getNullBits());
+    }
+
+    private void putValue(Object value) throws IOException {
+      if (runState == RUN_STATE.INIT) {
+        startNewRun(value);
+      } else {
+        if (lastValue.equals(value)) {
+          putRepeatValue(value);
+        } else {
+          putNonRepeatValue(value);
+        }
+      }
+    }
+
+    // when last row is reached, write out all collected data
+    private byte[] collectResult() throws IOException {
+      switch (runState) {
+        case REPEATED_RUN:
+          writeRunLength(valueCount);
+          writeRunValue(lastValue);
+          break;
+        case NONREPEATED_RUN:
+          writeRunLength(valueCount | 0x80000000);
```

Comment:
I think writing short is enough as of now. Because we are restricting page size in short only